metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1763,13 +1773,42 @@ class MetalsLanguageServer(
                 .applyEdit(new ApplyWorkspaceEditParams(workspaceEdit))
                 .asScala
             } yield ()
-          }
+          })
 
           futureOpt.getOrElse {
             languageClient.showMessage(Messages.InsertInferredTypeFailed)
             Future.unit
           }.withObjectValue
         }
+      case ServerCommands.ExtractMemberDefinition() =>
+        CancelTokens.future { token =>
+          val args = params.getArguments().asScala
+
+          val futureOpt = for {
+            (params, uri) <- textDocumentPosition(args)
+            data = ExtractMemberDefinitionData(uri, params)
+            result <- codeActionProvider.executeCommands(data, token)
```

Comment:
Let's return a future here, but make it `Future.failed()` instead of None. We should almost never have this situation so it's best to have it fail explicitely.

This way we could do:
```scala
val futureOpt = for {
  (params, uri) <- textDocumentPosition(args)
} yield {
  val data = ExtractMemberDefinitionData(uri, params)
  // future for
  for {
    result <- codeActionProvider.executeCommands(data, token)
    future <- languageClient.applyEdit(result.edits)
  } yield {
    result.goToLocation.foreach { location =>
      languageClient.metalsExecuteClientCommand(
        new ExecuteCommandParams(
          ClientCommands.GotoLocation.id,
          List(location: Object).asJava
        )
      )
    }
  }
}

```